plugins-root/check_dhcp.c

```diff
@@ -470,10 +470,6 @@ int send_dhcp_discover(int sock){
 	packet_xid=random();
 	discover_packet.xid=htonl(packet_xid);
 
-	/**** WHAT THE HECK IS UP WITH THIS?!?  IF I DON'T MAKE THIS CALL, ONLY ONE SERVER RESPONSE IS PROCESSED!!!! ****/
-	/* downright bizzarre... */
-	ntohl(discover_packet.xid);
```

Comment:
is there a test case for this one or is it no longer the case?

Reply:
there is not test case to my knowledge and I wouldn't know how to test that. In  theory this does not do anything.
Testing on my machine did not show any changes in behaviour, so I removed it. I mean, it cannot possibly do anything!!! (Or we really need to find out why it does that).
Not sure what problem they had back then, might just have been a timing problem.

Reply:
It comes from [this commit](https://github.com/monitoring-plugins/monitoring-plugins/commit/d50a9917ef38c44ac2067dba508cf2e690ff1412), a port to non-linux platform. At best it's an indication of a bug somewhere, and iirc those type of calls are often implemented as macros and could possibly have side effects. Without knowing what architecture was affected it'll be pretty hard to reproduce anyway :/

Reply:
BTW if you can at least test under different optimization levels (-Os, -O0, -O1, -O2, -O3) - sometimes bugs are hidden until specific optimisations are applied (or not applied). You can also run it with valgrind while testing as it may help catch some hidden issues too.

Reply:
Actually GH followed the move from `plugins/` to `plugins-root/`, but not the earlier move form `contrib/`, so it's [this commit](https://github.com/monitoring-plugins/monitoring-plugins/commit/44a321cb8a42d6c0ea2d96a1086a17f2134c89cc#diff-02bebc64297416b78f09d437e98879fccab6216970fcc30ec32cd86a0584552c).

It's the initial import by Ethan and check_dhcp was in `contrib/` even though it bears Ethan's copyright notice without any other authorship information. So TL;DR it's been there like forever...

Reply:
Someone suggested to me, that this thing might have been a compiler bug. Probably the compiler optimized that part out of the code and broke something with it.
That line tells the compiler, that this is actually something needed.